dali/pipeline/operator/builtin/external_source_test.cc

```diff
@@ -467,11 +467,69 @@ TEST(ExternalSourceTestNoInput, Throw) {
   vector<string> outputs = {"data_out_cpu"};
 
   exe->Build(&graph, outputs);
-  exe->RunCPU();
-  exe->RunMixed();
-  exe->RunGPU();
-  DeviceWorkspace ws;
-  ASSERT_THROW(exe->Outputs(&ws), std::runtime_error);
+  ASSERT_THROW(exe->RunCPU(), std::exception);
```

Comment:
can we narrow down the exception that is thrown?

Reply:
Unfortunately no. `RunCPU` throws both `runtime_error` and `logic_error`, so `exception` is first common parent